### DISCUSSION TITLE
How to use global setting to set cell color by column's IsReadOnly Propperty?

### DISCUSSION THREAD

**Discussion by nasterhe** [AvaloniaUI/Avalonia#17606] [Q&A]
I use this code,but didn't work.
`<Style Selector="DataGridCell">
  <Setter Property="Background" Value="{Binding IsReadOnly, RelativeSource={RelativeSource Mode=FindAncestor, AncestorType=DataGridColumn},Converter={StaticResource BrushConvert}}"/>
</Style>`

**Comment by tobyfirth** [comment]
I think [By Property Matching](https://docs.avaloniaui.net/docs/reference/styles/style-selector-syntax#by-property-match) is what you would want to use in the selector.

```xml
<Style Selector="DataGridCell[IsReadOnly=true]">
    <Setter Property="Background" Value="Red" />
</Style>
```

**Reply by nasterhe** [reply]
> I think [By Property Matching](https://docs.avaloniaui.net/docs/reference/styles/style-selector-syntax#by-property-match) is what you would want to use in the selector.
> 
> ```
> <Style Selector="DataGridCell[IsReadOnly=true]">
>     <Setter Property="Background" Value="Red" />
> </Style>
> ```

Thanks，But DataGridCell can't has IsReadOnly property,which property blong to DataGridColumn.
